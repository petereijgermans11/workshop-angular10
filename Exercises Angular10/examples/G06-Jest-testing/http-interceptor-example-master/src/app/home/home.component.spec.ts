import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { HomeComponent } from './home.component';
import { StatusService } from '../shared/service/status.service';
import { of } from 'rxjs';

const statusServiceMockFactory = () => ({
  getStatus$: of('OK')
});

describe('HomeComponent', () => {
  let component: HomeComponent;
  let fixture: ComponentFixture<HomeComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ HomeComponent ],
      providers: [ { provide: StatusService, useFactory: statusServiceMockFactory } ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(HomeComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  test('should create', () => {
    expect(component).toBeTruthy();
  });
});
