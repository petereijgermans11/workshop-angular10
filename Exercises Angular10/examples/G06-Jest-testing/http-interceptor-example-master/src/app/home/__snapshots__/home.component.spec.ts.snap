// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeComponent should render properly 1`] = `
<app-home
  statusService={[Function Object]}
  statusText$={[Function Observable]}
  triggerStatus$={[Function Subject]}
>
  <p>
    Home page
  </p><button>
    Get status
  </button><p>
    no status
  </p>
</app-home>
`;
